UndertaleModLib/Util/QoiConverter.cs

```diff
@@ -145,7 +170,16 @@ public unsafe static Bitmap GetImageFromStream(Stream s)
         /// <param name="padding">The amount of bytes of padding that should be used.</param>
         /// <returns>A QOI Image as a byte array.</returns>
         /// <exception cref="Exception">If there was an error with stride width.</exception>
-        public unsafe static byte[] GetArrayFromImage(Bitmap bmp, int padding = 4)
+        public static byte[] GetArrayFromImage(Bitmap bmp, int padding = 4) => GetSpanFromImage(bmp, padding).ToArray();
+
+        /// <summary>
+        /// Creates a QOI image as a span of bytes from a Bitmap.
+        /// </summary>
+        /// <param name="bmp">The bitmap to create the QOI image from.</param>
+        /// <param name="padding">The amount of bytes of padding that should be used.</param>
+        /// <returns>A QOI Image as a span of bytes.</returns>
+        /// <exception cref="Exception">If there was an error with stride width.</exception>
+        public unsafe static Span<byte> GetSpanFromImage(Bitmap bmp, int padding = 4)
         {
             byte[] res = new byte[(bmp.Width * bmp.Height * 4 * 12) + padding]; // default capacity
```

Comment:
BTW, didn't you want to change this, or do I remember wrong?

Reply:
Yeah, didn't you want to fix that in this PR?

Reply:
yeah i'm working on it rn